Minor cleanup/speedup to hacking/env-setup

bdowling opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- Feature Pull Request
- Docs Pull Request
##### COMPONENT NAME

<!--- Name of the plugin/module/task -->

hacking/env-setup
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.3.0 (fix_env-setup 26c4d6065c) last updated 2016/10/13 16:50:09 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 275fa3f055) last updated 2016/10/13 14:12:11 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 3e1ea76a75) last updated 2016/10/13 14:12:11 (GMT -400)
```
##### SUMMARY

<!--- Describe the change, including rationale and design decisions -->

Made it faster to source the hacking/env-setup by allowing gen_egg_info and removal of .pyc files to be run in the background.   I went down a path to make gen_egg dependent on file-timestamps as well, but backed out of that option.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

<!-- Paste verbatim command output below, e.g. before and after your change -->

```
Nothing to see here.
```

Allow egg-info creation (and .pyc removal) to happen in background when -q is
used to speedup sourcing hacking/env-setup.  Couple other minor fixes:
chdir back to parent was unnecessary due to already in subshell,
.\* on expr matches was extraneous.

Added doc info to make env-setup a bit more static by adding a source command
to your profile

bdowling committed:
Minor cleanup/speedup to hacking/env-setup

Allow egg-info creation (and .pyc removal) to happen in background when -q is
used to speedup sourcing hacking/env-setup.  Couple other minor fixes:
chdir back to parent was uneccessary due to already in subshell,
.* on expr matches was extraneous.

Added doc info to make env-setup a bit more static by adding a source command
to your profile

robinro commented:
@bdowling I like the idea, but this might break a setup that uses env-setup in some other script, since the files are not delted at the end of the command.

Better put `wait` at the end. At least things run in parallel then.

bdowling commented:
I had thought that might be a response to this, so we're thinking alike, that's good. ;)

What if I were to add another flag to optionally background?  Then you could pass -b -q for example?

This is also comes back to user's comfort level -- e.g. sourcing a file updated from github in .profile's vs just add the *PATH vars to their profile statically, the only thing they might be missing is the updated egg_info at times..

robinro commented:
`-b` sounds good to me

bdowling committed:
Added -b option to background the slower env-setup tasks

Still runs them in background, but will wait for the children to end
before returning.  This change causes it to act as previously in case
there are any CI or other type jobs that depend on it finishing before
doing next steps.  Updated docs to mention this usage.

mscherer commented on docsite/rst/intro_installation.rst:
```diff
@@ -140,10 +140,14 @@ Using Fish::
 
     $ source ./hacking/env-setup -q
 
+If you want to make ansible always avaialble in your environment, add it to your shell's profile (assumes you already source'd this above to set $ANSIBLE_HOME)::
```
'avaialble' is a typo.

bdowling committed:
Docs available typo

samdoran commented:
Thank you very much for your submission to Ansible.

Unfortunately, we're not sure if we want this feature in the program. Our reasons for this are:

   * This adds a lot of complexity without a lot of benefit
   * Optimization of this script isn't _really_ needed.

However, we're absolutely always up for discussion. Since this is a really busy project, we don't always see comments on closed tickets, but want to encourage open dialog. You can stop by the development list, and we'd be glad to talk about it - and we might even be persuaded otherwise!

   * https://groups.google.com/forum/#!forum/ansible-devel

In the future, sometimes starting a discussion on the development list prior to implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible.
